sdk/src/main/java/io/customer/sdk/util/SimpleTimer.kt

```diff
@@ -19,29 +26,58 @@ class AndroidSimpleTimer(
 ) : SimpleTimer {
 
     @Volatile private var countdownTimer: CountDownTimer? = null
+    @Volatile private var coroutineTimer: Deferred<*>? = null
     @Volatile private var timerAlreadyScheduled = false
     private val instanceIdentifier = String.random
 
+    override suspend fun <T> scheduleAndCancelPreviousSuspend(
+        seconds: Seconds,
+        block: suspend () -> T
+    ): T {
+        return coroutineScope {
+            // using deferred as it allows us to cancel our timer.
+            val newTimer: Deferred<T> = synchronized(this@AndroidSimpleTimer) {
+                timerAlreadyScheduled = true
```

Comment:
This is a bit confusing here, `deferred` coroutines can run in parallel and all of them will be using  `timerAlreadyScheduled` value. 

On top of that, we are using a `synchronized` block as well. 

So if we are not planning on running the coroutines in parallel, shouldn't we just run them synchronously?

Reply:
I removed the `suspend` function from the `SimpleTimer` so this code no longer exists. I replaced the logic with a simple `delay()` in the coroutine that performs HTTP requests.